code/modules/research/xenobiology/xenobiology.dm

```diff
@@ -447,6 +447,50 @@
 	if(loc == usr && loc.Adjacent(over_object))
 		afterattack(over_object, usr, TRUE)
 
+/obj/item/slimepotion/oil_slick
+	name = "slime oil potion"
+	desc = "A potent chemical mix that will remove the slowdown from any item by reducing friction. Doesn't mix well with water."
+	icon = 'icons/obj/chemical.dmi'
+	icon_state = "bottle4"
+	origin_tech = "biotech=5"
+
+/obj/item/slimepotion/oil_slick/afterattack(obj/O, mob/user, proximity_flag)
+	if(!proximity_flag)
+		return
+	..()
+	if(!istype(O))
+		to_chat(user, "<span class='warning'>The potion can only be used on items or vehicles!</span>")
+		return
+	if(isitem(O))
```

Comment:
You can simplify this by typecasting obj/O as obj/item/O and just using a single istype, no var reassignment later as well

Reply:
We need to check for vehicles as well, which are not items

Reply:
> We need to check for vehicles as well, which are not items

You can put the vehicle check after a failed istype check

Reply:
Typecasted to item, and table